packages/medusa/src/api-v2/admin/price-lists/[id]/prices/batch/add/route.ts

```diff
@@ -0,0 +1,38 @@
+import { createPriceListPricesWorkflow } from "@medusajs/core-flows"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../../../types/routing"
+import { listPriceLists } from "../../../../queries"
+import {
+  adminPriceListRemoteQueryFields,
+  defaultAdminPriceListFields,
+} from "../../../../query-config"
+import { AdminPostPriceListsPriceListPricesBatchAddReq } from "../../../../validators"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<AdminPostPriceListsPriceListPricesBatchAddReq>,
+  res: MedusaResponse
+) => {
+  const { prices } = req.validatedBody
+
+  const id = req.params.id
+  const workflow = createPriceListPricesWorkflow(req.scope)
+  const { errors } = await workflow.run({
+    input: { data: [{ id, prices }] },
+    throwOnError: false,
+  })
+
+  if (Array.isArray(errors) && errors[0]) {
+    throw errors[0].error
+  }
+
+  const [[priceList]] = await listPriceLists({
+    container: req.scope,
+    remoteQueryFields: adminPriceListRemoteQueryFields,
+    apiFields: defaultAdminPriceListFields,
+    variables: { filters: { id }, skip: 0, take: 1 },
```

Comment:
todo: why do we have pagination here?

Reply:
meaning? I'm just doing a take 1

Reply:
Will keep an eye on this comment to see if there is anything to change. 🤞🏻

Reply:
I mean, there should not be a need to do take one no?

Reply:
I just don't see the need to do pagination to force a take 1 here ahah, since you are filtering by id it can just do a retrieve no?

Reply:
Ah, I just noticed that the remote query does a retrieve in that case. Didn't know that 😄 thanks! Will be added in the base PR

Reply:
ahah no worries, now you know :)